### PR TITLE
Add MCP ToolProvider boundary behind deterministic descriptor registry

### LINKED ISSUE #541: task: add MCP ToolProvider boundary behind deterministic descriptor registry
## Context

The current tool contract has descriptor-based tool execution, validation, deterministic mock behavior, and gated real vault append support. It explicitly does not ship MCP ToolProvider integration, dynamic discovery, or remote server multiplexing. The next bounded follow-up is a ToolProvider boundary that uses the existing descriptor registry as authority and keeps deterministic behavior intact.

## Scope

- Add an MCP ToolProvider abstraction/boundary that exposes currently registered descriptors through a provider interface.
- Keep descriptor registry YAML as the source of truth for available tools; do not add dynamic remote discovery.
- Ensure provider calls route through existing validation, policy, timeout, and mock/real execution paths.
- Add tests that provider-backed execution is equivalent to current descriptor-backed execution for at least one mock tool and the gated vault append path.
- Update the tool policy contract and architecture docs to describe the shipped provider boundary and preserve non-claims.

## Source Anchors

- `docs/ROADMAP.md` :: Post-v5.6 follow-ups / A2A/MCP orchestration routing
- `docs/ARCHITECTURE.md` :: Runtime posture table / MCP/tools
- `docs/ARCHITECTURE.md` :: Post-v5.6 Follow-ups (Non-baseline)
- `docs/contracts/TOOL_POLICY_AND_MCP_ADAPTER_CONTRACT.md` :: Future MCP integration boundary
- `docs/contracts/TOOL_POLICY_AND_MCP_ADAPTER_CONTRACT.md` :: Current posture

## Constraints

- Do not implement remote MCP server multiplexing or dynamic tool discovery.
- Do not bypass descriptor validation, policy enforcement, or timeout handling.
- Do not change existing tool ids or descriptor format unless necessary and documented.
- Preserve deterministic mock execution for CI.
- Keep real vault append gated by existing allowlist/feature flag behavior.

## Acceptance Criteria

- [ ] A ToolProvider boundary exposes registered tool descriptors without changing registry authority.
  Verify: `tests/tools/test_mcp_tool_provider.py::test_tool_provider_lists_registry_descriptors`
- [ ] Provider-backed mock tool execution matches current executor behavior and validation errors.
  Verify: `tests/tools/test_mcp_tool_provider.py::test_tool_provider_mock_execution_matches_descriptor_executor`
- [ ] Provider-backed vault append remains gated by allowlist and feature flags.
  Verify: `tests/tools/test_mcp_tool_provider.py::test_tool_provider_vault_append_respects_existing_gates`
- [ ] Docs describe the shipped provider boundary while keeping remote multiplexing and dynamic discovery out of scope.
  Verify: doc writeback at `docs/contracts/TOOL_POLICY_AND_MCP_ADAPTER_CONTRACT.md :: Future MCP integration boundary` and `docs/ARCHITECTURE.md :: Runtime posture table / MCP/tools`

## Out of Scope

- Remote MCP server multiplexing.
- Dynamic tool discovery.
- Descriptor versioning redesign.
- New production tool permissions model.
- A2A transport changes.

## Suggested Validation

- `PYTEST_DISABLE_PLUGIN_AUTOLOAD=1 pytest -q tests/tools/test_mcp_tool_provider.py tests/orchestrator/test_executor_tool_validation.py`
- `PYTEST_DISABLE_PLUGIN_AUTOLOAD=1 pytest -q tests/settings/test_tools_loader.py tests/architecture/test_tool_contracts.py`
- `rg -n "ToolProvider|dynamic discovery|remote MCP|descriptor registry" docs/contracts/TOOL_POLICY_AND_MCP_ADAPTER_CONTRACT.md docs/ARCHITECTURE.md`

## Source Docs

- `docs/ROADMAP.md`
- `docs/ARCHITECTURE.md`
- `docs/contracts/TOOL_POLICY_AND_MCP_ADAPTER_CONTRACT.md`
- `docs/contracts/A2A_CONTRACT_AND_TRACE.md`

### PR DESCRIPTION
**Pull request by RasmusTho**
Fixes #541

## Summary
- add a registry-backed `McpToolProvider` boundary that lists and executes tools via the existing executor path
- preserve descriptor registry authority, validation/policy/timeout behavior, and gated vault append semantics
- add focused tests and doc writeback describing shipped provider boundary and explicit non-claims

## Validation
- `PYTEST_DISABLE_PLUGIN_AUTOLOAD=1 pytest -q tests/tools/test_mcp_tool_provider.py`
- `rg -n "ToolProvider|dynamic discovery|remote MCP|descriptor registry" docs/contracts/TOOL_POLICY_AND_MCP_ADAPTER_CONTRACT.md docs/ARCHITECTURE.md`

---

### CODE REVIEW

File: app/orchestrator/mcp_tool_provider.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+        step = PlanStep(
+            id=step_id,
+            kind="tool_call",
+            description=description,
+            tool=tool_name,
+            tool_args=dict(tool_args or {}),
+        )
+        active_executor = executor or MockPlanExecutor()
+        return active_executor.execute_step(step, context)
```
[the remark this answers is not in the document]
Comment (RasmusTho) [reply app/orchestrator/mcp_tool_provider.py]: Addressed in 6e9bb3c: provider discovery now advertises only executor-supported registry tools, and execution uses the provider-resolved descriptor path with executor parity validation/invocation.
